tests/topotests/bgp-snmp-mplsl3vpn/test_bgp_snmp_mplsvpn.py

```diff
@@ -0,0 +1,730 @@
+# WHETHER IN AN ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS
+# ACTION, ARISING OUT OF OR IN CONNECTION WITH THE USE OR PERFORMANCE
+# OF THIS SOFTWARE.
+#
+
+"""
+test_bgp_snmp_mplsl3vpn.py: Test mplsL3Vpn MIB [RFC4382].
+"""
+
+import os
+import sys
+import json
+from functools import partial
+from time import sleep
+import pytest
+
+# Save the Current Working Directory to find configuration files.
+CWD = os.path.dirname(os.path.realpath(__file__))
+sys.path.append(os.path.join(CWD, "../"))
+
+# pylint: disable=C0413
+# Import topogen and topotest helpers
+from lib import topotest
+from lib.topogen import Topogen, TopoRouter, get_topogen
+from lib.topolog import logger
+from lib.snmptest import SnmpTester
+
+# Required to instantiate the topology builder class.
+from mininet.topo import Topo
+
+
+class TemplateTopo(Topo):
+    "Test topology builder"
+
+    def build(self, *_args, **_opts):
+        "Build function"
+        tgen = get_topogen(self)
+
+        # This function only purpose is to define allocation and relationship
+        # between routers, switches and hosts.
+        #
+        #
+        # Create routers
+        tgen.add_router("r1")
+        tgen.add_router("r2")
+        tgen.add_router("r3")
+        tgen.add_router("r4")
+        tgen.add_router("ce1")
+        tgen.add_router("ce2")
+        tgen.add_router("ce3")
+        tgen.add_router("ce4")
+
+        # r1-r2
+        switch = tgen.add_switch("s1")
+        switch.add_link(tgen.gears["r1"])
+        switch.add_link(tgen.gears["r2"])
+
+        # r1-r3
+        switch = tgen.add_switch("s2")
+        switch.add_link(tgen.gears["r1"])
+        switch.add_link(tgen.gears["r3"])
+
+        # r1-r4
+        switch = tgen.add_switch("s3")
+        switch.add_link(tgen.gears["r1"])
+        switch.add_link(tgen.gears["r4"])
+
+        # r1-ce1
+        switch = tgen.add_switch("s4")
+        switch.add_link(tgen.gears["r1"])
+        switch.add_link(tgen.gears["ce1"])
+
+        # r1-ce3
+        switch = tgen.add_switch("s5")
+        switch.add_link(tgen.gears["r1"])
+        switch.add_link(tgen.gears["ce3"])
+
+        # r1-ce4
+        switch = tgen.add_switch("s6")
+        switch.add_link(tgen.gears["r1"])
+        switch.add_link(tgen.gears["ce4"])
+
+        # r1-dangling
+        switch = tgen.add_switch("s7")
+        switch.add_link(tgen.gears["r1"])
+
+        # r2-r3
+        switch = tgen.add_switch("s8")
+        switch.add_link(tgen.gears["r2"])
+        switch.add_link(tgen.gears["r3"])
+
+        # r3-r4
+        switch = tgen.add_switch("s9")
+        switch.add_link(tgen.gears["r3"])
+        switch.add_link(tgen.gears["r4"])
+
+        # r4-ce2
+        switch = tgen.add_switch("s10")
+        switch.add_link(tgen.gears["r4"])
+        switch.add_link(tgen.gears["ce2"])
+
+
+def setup_module(mod):
+    "Sets up the pytest environment"
+
+    # skip tests is SNMP not installed
+    if not os.path.isfile("/usr/sbin/snmpd"):
```

Comment:
Is it always at this location? What if snmpd isn't installed from package, but locally (and ends up in `/usr/local/sbin`)?

Maybe better to run the `which snmpd` command and see if the system can find it?

Reply:
done